src/cheri/cheri-pte-ext.adoc

```diff
@@ -7,40 +7,38 @@ endif::[]
 
 NOTE: {cheri_priv_crg_ext} depends on Supervisor-Level ISA and {cheri_base64_ext_name} also being supported.
 
 NOTE: {cheri_priv_crg_ext} is strongly recommended but not mandatory as a future version of this specification may specify an improved method.
 
-NOTE: There is no explicit mechanism for enabling or disabling {cheri_priv_crg_ext}.
-
-NOTE: If software ignores the new PTE bits then there is no change in functionality unless capabilities are accessed.
-
 NOTE: A future version of this specification may include kernel revocation which may require an <<sstatusreg_pte,sstatus>>.SCRG bit.
 
 The minimum level of PTE support is to set CRW and CD to 1 in all PTEs intended for storing
 capabilities (i.e., private anonymous mappings) and leave
 <<sstatusreg_pte,sstatus>>.UCRG and CRG in all PTEs set to 0, which will allow
-capabilities with their tags set to be loaded and stored successfully.
+capabilities with their {ctag}s set to be loaded and stored successfully.
+
+NOTE: When {cheri_priv_crg_ext} is enabled (<<sstatusreg_pte,sstatus>>.CRGE is set), the semantics of PTE.CRW bit are redefined relative to the definition in <<section_priv_cheri_vmem>>.
+The ability to load and store capabilities in the page depends on all of the PTE.CRW, PTE.CD, PTE.CRG bits, which effectively become a single 3-bit field controlling the behavior.
+Therefore, when PTE.CRW is clear, it is not required that future extensions to {cheri_priv_crg_ext}, such as for kernel revocation, cannot store capabilities to a page.
 
 NOTE: Hardware-initiated memory accesses from the page-table walker are not checked by a capability.
 
 [#cap_revocation]
 === Capability Revocation
 
-Page table enforcement can accelerate concurrent capability revocation
-for temporal safety.
-Without page table capability protection, a concurrent capability revocation
-sweep must begin by visiting all PTEs to mark them unreadable, henceforth trapping on any
-read to a new page to sweep it clean before proceeding.
-With a page-granularity generational capability read permission, we can
-eliminate the initial permission change of all PTEs.
-In addition, a page-granularity capability write control can eliminate many pages
-from the sweep that are known to not contain capabilities.
-Page-granularity capability-dirty tracking accelerates the detection of capability-clean
-pages, which enables the revocation sweep to skip them during the visit.
+cite:[cornucopia-reloaded] details algorithms for preventing use-after-free in virtual memory, and non-virtual memory based systems.
+Revocation is the process of clearing the {ctag} of capabilities which point to freed memory, to prevent them from being dereferenced.
+A CHERI system which implements such a scheme is said to support _temporal safety_.
```

Comment:
```suggestion
Revocation is the process of clearing the {ctag} of capabilities that point to freed memory, to prevent them from being dereferenced and thereby enforcing _temporal memory safety_.
cite:[cornucopia-reloaded] details algorithms for preventing use-after-free in virtual memory, and non-virtual memory based systems.
```